lib/nerves_hub_device/socket.ex

```diff
@@ -83,20 +83,25 @@ defmodule NervesHubDevice.Socket do
 
       {:ok, i2c} = ATECC508A.Transport.I2C.init(bus_name: "i2c-#{bus_num}")
 
-      {:ok, engine} = NervesKey.PKCS11.load_engine()
+      if NervesKey.provisioned?(i2c) do
+        {:ok, engine} = NervesKey.PKCS11.load_engine()
 
-      cert =
-        NervesKey.device_cert(i2c, certificate_pair)
-        |> X509.Certificate.to_der()
+        cert =
+          NervesKey.device_cert(i2c, certificate_pair)
+          |> X509.Certificate.to_der()
 
-      signer_cert =
-        NervesKey.signer_cert(i2c, certificate_pair)
-        |> X509.Certificate.to_der()
+        signer_cert =
+          NervesKey.signer_cert(i2c, certificate_pair)
+          |> X509.Certificate.to_der()
 
-      key = NervesKey.PKCS11.private_key(engine, i2c: bus_num)
-      cacerts = [signer_cert | cacerts]
+        key = NervesKey.PKCS11.private_key(engine, i2c: bus_num)
+        cacerts = [signer_cert | cacerts]
 
-      [cert: cert, key: key, cacerts: cacerts]
+        [cert: cert, key: key, cacerts: cacerts]
+      else
+        Logger.error("[NervesHubDevice] NervesKey isn't provisioned, so not using.")
+        []
```

Comment:
It would be nice to do the same thing here as in the `else` catch all and slightly adjust the error to say “attempting user configured options” rather than “so not using” just have _something_

But not a block to me so your call. This will work fine as is to get us moving